opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3.go

```diff
@@ -0,0 +1,139 @@
+package opentelekomcloud
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/opentelekomcloud/gophertelekomcloud/openstack/dds/v3/flavors"
+)
+
+func dataSourceDDSFlavorV3() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDdsFlavorV3Read,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"engine_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"mongos", "shard", "config", "replica",
+				}, true),
+			},
+			"vcpus": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"memory": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"flavors": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"spec_code": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"vcpus": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"memory": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"az_status": {
+							Type:     schema.TypeMap,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceDdsFlavorV3Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ddsClient, err := config.ddsV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating OpenTelekomCloud DDS client: %s", err)
+	}
+
+	listOpts := flavors.ListOpts{
+		Region:     GetRegion(d, config),
+		EngineName: d.Get("engine_name").(string),
+	}
+
+	pages, err := flavors.List(ddsClient, listOpts).AllPages()
+	if err != nil {
+		return fmt.Errorf("unable to all flavor pages: %s", err)
+	}
+
+	flavorsList, err := flavors.ExtractFlavors(pages)
+	if err != nil {
+		return fmt.Errorf("unable to extract flavors: %s", err)
+	}
+
+	flavorList := make([]map[string]interface{}, 0)
```

Comment:
`flavorList` and `flavorsList` are too similar, please rename variables to make it easier to distinguish between them